tests/harvester/test_mets.py

```diff
@@ -93,3 +93,46 @@ def test_files_exception_on_two_locations_for_a_file(
     with pytest.raises(METSLocationParseError):
         for _ in mets.files():
             pass
+
+
+def test_file_content_type_parsing(simple_mets_path, mets_dc_identifier):
+    """
+    Test content type parsing when there's one location for each file.
+    """
+    mets = METS(simple_mets_path, mets_dc_identifier)
+    files = list(mets.files())
+
+    first_file = files[0]
+    assert first_file.filetype == "UnknownTypeFile"
+
+    last_file = files[-1]
+    assert last_file.filetype == "ALTOFile"
+
+
+def test_alto_files(simple_mets_path, mets_dc_identifier):
+    """
+    Ensure that an accurate list of alto files is returned.
+    """
+    mets = METS(simple_mets_path, mets_dc_identifier)
+    alto_files = list(mets.files_of_type("ALTOFile"))
+    assert len(alto_files) == 4
+    assert all(file.filetype == "ALTOFile" for file in alto_files)
+
+
+def test_download_alto_files(tmp_path, simple_mets_path, mocker, mets_dc_identifier):
+    """
+    Test downloading all ALTO files listed in a METS file.
+
+    This is done by checking that file.download is called the correct number of times
+    during a download_alto_files call.
+    """
+    mets = METS(simple_mets_path, mets_dc_identifier)
+    mocker.patch("harvester.file.File.download")
+    mocker.patch(
+        "harvester.mets.METS.files_of_type", return_value=(File for f in range(4))
```

Comment:
Do we need this patch? With it, we would be none the wiser if `download_alto_files` tried to download all images instead (okay, currently we would be, but as soon as they are no longer UnknownFile we wouldn't).

It would be great if we can figure out a way of ensuring that the "downloaded" files are indeed the ALTOs. But if we can't, or it's way too much work, then this is totally fine.

Reply:
Good catch! The patch is in fact no longer needed after the latest changes and it needs to be removed.

I'm not sure how to check the contents of a downloaded file without actually downloading it. The function that extracts ALTO files checks that the file type is ALTO, and the same function is used for the downloading, so I wonder if that would be enough assurance that the files are actually ALTOs. 

Thoughts?

Reply:
The biggest risk here might be that Someone™ (i.e. me) adds e.g. `download_preservation_images()` by copypasting the existing function but forgets to change the type of file downloaded. At least that we can catch if we only patch `harvester.file.ALTOFile.download`, as trying to call download on any other file type should raise an error when the test suite tries to make forbidden http requests.

A heavier but also more accurate option would be to add a side effect to the mocked function and do some checks using that, e.g. recording the class of the object from which the call was made. But that's quite a hassle and I'm not entirely sure if the mock even has access to `self`, so I'd say that this isn't worth it in this case.

Reply:
Good idea, only patching `harvester.file.ALTOFile.download` does the job! :) Fixed now.

Reply:
Yay, looking good!